modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.ingest.meta.txt

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the Server Side Public License, v 1; you may not use this file except
+# in compliance with, at your election, the Elastic License 2.0 or the Server
+# Side Public License, v 1.
+#
+
+# This file only whitelists the script class and the Metadata class for ingest scripts.
+# Due to name conflicts, this whitelist cannot be loaded with other scripts with conflicting Metadata types.
+
+class org.elasticsearch.script.IngestScript$Metadata @alias[class="Metadata"] {
+    String getIndex()
+    void setIndex(String)
+    String getId()
+    void setId(String)
+    String getRouting()
+    void setRouting(String)
+    long getVersion()
+    void setVersion(long)
+    void removeVersion()
+    boolean hasVersion()
+    VersionType getVersionType()
+    void setVersionType(VersionType)
+    ZonedDateTime getTimestamp()
+}
+
+class org.elasticsearch.script.IngestScript {
+    Metadata meta()
+}
+
+class org.elasticsearch.index.VersionType {
```

Comment:
Yes, `VersionType.INTERNAL`.

We could possibly bring them in as constants on the script class, which should make that possible.  It would impact any variable named `internal`, however.

Reply:
We could make those available as getters on the script object but that would be a back compat issue with existing variables of the same name.  Scripts would get:

> variable [external] is already defined

We'd have to have a way to shadow to avoid that issue.

Reply:
Changed to taking string.